### PR TITLE
docs: add docstrings returns section to `bigframes._confi`

### PR DESCRIPTION
**Pull request by arwas11**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #393420281 🦕

### CODE REVIEW

File: bigframes/_config/__init__.py [hunk 1]
```diff
@@ -109,6 +128,11 @@ def is_bigquery_thread_local(self) -> bool:
 
         A thread-local session can be started by using
         `with bigframes.option_context("bigquery.some_option", "some-value"):`.
+
+        Returns:
+            Boolean:
```
Comment (tswast) [inline bigframes/_config/__init__.py:133]: This type is called `bool` in Python.

**Review by tswast — COMMENTED** [review]
Looking great!

File: bigframes/_config/__init__.py [hunk 2]
```diff
@@ -95,12 +105,21 @@ def sampling(self) -> sampling_options.SamplingOptions:
         (e.g., to_pandas, to_numpy, values) or implicitly (e.g.,
         matplotlib plotting). This option can be overriden by
         parameters in specific functions.
+
+        Returns:
+            bigframes._config.sampling_options.SamplingOptions:
+              Options for controlling downsampling.
```
Comment (tswast) [inline bigframes/_config/__init__.py:111]: Nit: Should be indented another two spaces. We use multiples of 4 spaces for indentation.

```suggestion
                Options for controlling downsampling.
```

File: bigframes/_config/__init__.py [hunk 3]
```diff
@@ -95,12 +105,21 @@ def sampling(self) -> sampling_options.SamplingOptions:
...
         return self._local.sampling_options
 
     @property
     def compute(self) -> compute_options.ComputeOptions:
-        """Thread-local options controlling object computation."""
+        """Thread-local options controlling object computation.
+
+        Returns:
+            bigframes._config.compute_options.ComputeOptions:
+              Thread-local options for controlling object computation
```
Comment (tswast) [inline bigframes/_config/__init__.py:121]: Same here (2 more spaces)

File: bigframes/_config/__init__.py [hunk 4]
```diff
@@ -109,6 +128,11 @@ def is_bigquery_thread_local(self) -> bool:
 
         A thread-local session can be started by using
         `with bigframes.option_context("bigquery.some_option", "some-value"):`.
+
+        Returns:
+            bool:
+              A boolean value, where a value is True if a thread-local session
```
Comment (tswast) [inline bigframes/_config/__init__.py:134]: Same here and the line below (2 more spaces)

File: bigframes/_config/__init__.py (hunk 3 above)
Comment (arwas11) [reply bigframes/_config/__init__.py]: Done.

File: bigframes/_config/__init__.py (hunk 4 above)
Comment (arwas11) [reply bigframes/_config/__init__.py]: Done.

**Review by tswast — APPROVED** [review]

**Pull request merged**